lib/eventsource.js

```diff
@@ -170,21 +254,21 @@ function EventSource (url, eventSourceInitDict) {
       }
 
       if (res.statusCode !== 200) {
-        _emit('error', new Event('error', {status: res.statusCode, message: res.statusMessage}))
+        _emit('error', new Event('error', { status: res.statusCode, message: res.statusMessage }))
         return self.close()
       }
 
       readyState = EventSource.OPEN
       res.on('close', function () {
         res.removeAllListeners('close')
         res.removeAllListeners('end')
-        onConnectionClosed()
+        onConnectionFailed()
```

Comment:
If the user called the `close` function in the SDK to close the connection, won't this fire an error? Same with `end` below?

Reply:
Good catch, let me verify quickly

Reply:
Just checked and it won't. 

The Node SDK's `close()` function calls this lib's `_close()` function which puts the `readyState` to `CLOSED` - so `onConnectionFailed` will return early if it detects that the readyState is `CLOSED` 

What do you think?

Reply:
function onConnectionFailed (error) {
    if (readyState === EventSource.CLOSED) {
      return
    }

...
}

Reply:
For `end` - yes it will be reported as an error.

Reply:
Chatted with you offline on this. You are happy that `close()` won't fire an error. And for `end` we just emit this as an error, as FF-Server should never end a connection, and if it does it's defo an error